### ISSUE TITLE
API changes towards EzXML.jl 1.0 on Julia 1.0.

### ISSUE THREAD

**Issue by bicycle1885** [bicycle1885/EzXML.jl#51]
Some of these changes will need features introduced in Julia 1.0.

## Properties

- Add node properties: `type`, `path`, `name`, `content`, `namespace`.
- Add document properties: `root`, `dtd`.
- Add tree traversal properties: `document`, `parentnode`, `firstelement`, `lastelement`, `nextelement`, `prevelement`, `firstnode`, `lastnode`, `nextnode`, `prevnode`.

All these properties return a `nothing` when it has no corresponding property.

## XPath

- Rename `find` to `findall`.
- Make `find*` function take an XPath query as the first argument.
- Make `findfirst` and `findlast` return `nothing` when there is no such node.

This is because the interface of `find*` has changed in Julia 1.0.

## StreamReader

- Introduce the `ReaderEvent` type.
- Add some properties to `ReaderEvent`: `type`, `depth`, `name`, `content`, `namespace`.
- Add an attribute accessor to `ReaderEvent` (e.g. `event["attr"]`)
- Iterate over reader events; not over `ReaderType`s.

**Comment by ararslan** [comment]
Will the accessor functions be deprecated in favor of properties? 😟

**Comment by bicycle1885** [comment]
Yes, I plan to do that because properties do not import so many names into the current module. Any objections against it?

**Comment by ararslan** [comment]
Yes, I'm very, very much against that. The names this package uses are unlikely to create any conflict, so I don't see why exporting a lot of things would matter.

**Comment by ararslan** [comment]
I disagree that it looks at all better; I think it looks much less like Julia and more like Python. Things like `primates.firstelement.prevelement` were what I was most worried about proliferating once `getproperty` was introduced.

**Comment by bicycle1885** [comment]
I disagree with you. `node.firstelement.prevelement` looks better than `prevelement(firstelement(node))` to me because you don't need to read/write the code from left to right in this case. I think method chaining is not a Julian way but property chaining just looks so good as field chaining as we already do when handling nested structs.

**Comment by omus** [comment]
I'll be taking care of some of this in order to get EzXML working on Julia 1.0
